### PR TITLE
test/rbd: cli_generic fails if v1 image format or deep-flatten disabled

### PR DESCRIPTION
**Pull request by Songweibin**
…deep-flatten disabled

fix: http://tracker.ceph.com/issues/22950

Signed-off-by: songweibin <song.weibin@zte.com.cn>

### CODE REVIEW

**Comment by Songweibin** [comment]
@dillaman  @trociny  Thank you for those tests.

File: qa/workunits/rbd/cli_generic.sh [hunk 1]
```diff
@@ -93,6 +93,8 @@ test_others() {
     rbd info testimg5 | grep 'size 256 MB'
     rbd snap ls testimg5 | grep -v 'SNAPID' | wc -l | grep 1
     rbd snap ls testimg5 | grep '.*snap2.*'
+    rbd snap rm testimg4@snap2
+    rbd snap rm testimg5@snap2
```
Comment (trociny) [inline qa/workunits/rbd/cli_generic.sh]: Just wonder, don't we want to make this conditional, i.e. remove only if deep-flatten is not enabled? Something like below:
```
rbd info testimg4 | grep 'features:.*deep-flatten' || rbd snap rm testimg4@snap2
...
```

Comment (dillaman) [reply qa/workunits/rbd/cli_generic.sh]: :+1:

**Comment by Songweibin** [comment]
@trociny  Updated, thanks.
I think the tests for deep copy clone (`test_others` function) is already called after RBD_CREATE_ARGS="--image-format 2"?

**Comment by dillaman** [comment]
@Songweibin The issue is that the images used during that test are not using ```RBD_CREATE_ARGS``` (a la [1])

[1] https://github.com/Songweibin/ceph/blob/ef72ae3e7e825934e020c0bbdb1a613f3bcf34ea/qa/workunits/rbd/cli_generic.sh#L90

**Comment by trociny** [comment]
`test_others` is called twice: both after `RBD_CREATE_ARGS=""` and after `RBD_CREATE_ARGS="--image-format 2"` and the first call will fail on `snap protect` if ceph.conf has `rbd default format = 1`.

@dillaman `RBD_CREATE_ARGS` for `rbd clone` probably good but not necessary in that case because I think it will inherit format from testimg1, which was created with `RBD_CREATE_ARGS`.

**Comment by trociny** [comment]
Probably we would want leave "deep copies" test in test_other so it would called for both v1 and v2 formats, and move "deep copy clone-image" test to "test_deep_copy_clone", called only for v2?

**Comment by dillaman** [comment]
@trociny Yeah, sounds good to me -- move it or wrap it w/ a conditional block

**Comment by Songweibin** [comment]
@dillaman  @trociny  Updated as suggested. Thanks for explaining this.

**Comment by trociny** [comment]
@Songweibin Please change in the commit log message 'fix:' to 'Fixes:'.
Otherwise LGTM.

**Comment by Songweibin** [comment]
@trociny Done, thanks.

**Pull request merged**